File: dart/lib/src/longitude.dart
/* This is free and unencumbered software released into the public domain. */

import 'angle.dart' show Angle;

/// A longitude.
///
/// See: https://dogma.dev/Longitude
class Longitude extends Angle {
  static const int minDegrees = -180;
  static const int maxDegrees = 180;

  const Longitude(final double degrees)
      : assert(degrees >= minDegrees),
        assert(degrees <= maxDegrees),
        super.fromDegrees(degrees);
}

File: js/src/longitude.ts
/* This is free and unencumbered software released into the public domain. */

/// <reference path="angle.ts"/>
export class Longitude extends Angle {
  public static readonly MIN_DEGREES: number = -180;
  public static readonly MAX_DEGREES: number = 180;

  public constructor(degrees: number) {
    if (degrees < Longitude.MIN_DEGREES) {
      throw new RangeError(`Longitude bounds are ±180°, but got ${degrees}°`);
    }
    if (degrees > Longitude.MAX_DEGREES) {
      throw new RangeError(`Longitude bounds are ±180°, but got ${degrees}°`);
    }
    super(degrees / 180 * Math.PI);
  }
}
